MovieApp/MovieCategoryViewController.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  MovieListViewController.swift
+//  MovieApp
+//
+//  Created by endava-bootcamp on 25.04.2023..
+//
+
+import Foundation
+import UIKit
+import PureLayout
+import MovieAppData
+
+protocol Coordinator {
+
+    func start()
+}
+
+class MovieCategoryViewController: UIViewController {
+    
+    var tableView: UITableView!
+    let movies = MovieUseCase().allMovies
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationItem.title = "Movie List"
+        
+        view.isUserInteractionEnabled = true
+
+        buildViews()
+    }
+    
+    @objc func collectionCellClicked() {
+//        self.navigationController?.popViewController(animated: true)
+        
+//        let vc = MovieDetailsViewController()
+//        navigationController?.pushViewController(vc, animated: true)
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        tableView.frame = view.bounds
+    }
+    
+    private func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+    
+    private func createViews() {
+        tableView = UITableView()
+        view.addSubview(tableView)
+
+    }
+    
+    private func styleViews() {
+        tableView.register(MovieCategoryCollection.self, forCellReuseIdentifier: MovieCategoryCollection.identifier)
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorStyle = .none
+        
+    }
+    private func defineLayoutForViews() {
+        tableView.autoPinEdge(toSuperviewSafeArea: .top)
+        tableView.autoPinEdge(toSuperviewSafeArea: .bottom)
+        tableView.autoPinEdge(toSuperviewSafeArea: .leading)
+        tableView.autoPinEdge(toSuperviewSafeArea: .trailing)
+    }
+}
+
+extension MovieCategoryViewController:UITableViewDataSource, UITableViewDelegate {
+        func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+            return 3
```

Comment:
Regarding the previous comment. if you had an array of movie collections you can just write `movieCollections.count`

This will be future proofing. Down the line maybe there will be more than three categories